consensus/src/config.rs

```diff
@@ -72,6 +72,14 @@ pub struct Config {
     pub max_round: Round,
 }
 
+impl Config {
+    /// Reaching a high round can be expected to introduce some slowdown.
+    /// Use this function to ensure that the slowdown will happen.
+    pub fn reaching_max_round_takes_at_least(&self, duration: Duration) -> bool {
```

Comment:
Since this is optional to use, it's very easy to skip or forget this check.

The bug we had in aleph-node was actually that we prepared a correct config, but in a dead code, and then ended up using a default one, and not the prepared one. And I'm pretty sure that it's possible to repeat this bug, by not being careful...

What I would propose to do, is to either:
1) Make sure that this check is always done, for instance in a constructor of `Config`, or
2) Introduce a new type `VerifiedConfig` which you get from `Config` by applying this check (and perhaps in the future also different ones), and make AlephBFT initialization expect `VerifiedConfig`.

Reply:
Out of these I prefer 1. since it doesn't create an additional type, although it will make the config annoyingly opaque compared to others – the fields will probably no longer be `pub`, since we don't want it to be constructable without the check.

On the other hand I hate `pub` fields, so maybe I should take this as a win. ;p